pkg/v1/config/clientconfig.go

```diff
@@ -22,6 +22,27 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 )
 
+func init() {
+	// Acquire tanzu config lock
+	AcquireTanzuConfigLock()
+	defer ReleaseTanzuConfigLock()
+
+	c, err := GetClientConfig()
+	if err != nil {
+		log.Warningf("unable to get client config: %v", err)
+	}
+
+	addedDefaultDiscovery := populateDefaultStandaloneDiscovery(c)
+	addedFeatureFlags := addDefaultFeatureFlagsIfMissing(c, DefaultCliFeatureFlags)
+	addedEdition := addDefaultEditionIfMissing(c)
+	addedBomRepo := addBomRepoIfMissing(c)
+	addedCompatabilityFile := addCompatibilityFileIfMissing(c)
+
+	if addedFeatureFlags || addedDefaultDiscovery || addedEdition || addedCompatabilityFile || addedBomRepo {
+		_ = StoreClientConfig(c)
+	}
```

Comment:
Instead of doing this update on each `GetClientConfig` call, do it as part of `init` call. The `GetClientConfig` should only be responsible for getting the configuration and not updating it.